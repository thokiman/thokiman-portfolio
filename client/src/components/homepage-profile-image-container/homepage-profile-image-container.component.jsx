import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectIsSideBarHidden } from "../../redux/header/header.selectors";

import {
  HomepageBackgroundProfileImage,
  HomepageProfileImage,
  HomepageProfileImageContainer,
} from "./homepage-profile-image-container.styles";

export const HomePageProfileImageContainer = ({ isSideBarHidden }) => {
  return (
    <HomepageProfileImageContainer $issidebarhidden={isSideBarHidden}>
      <HomepageBackgroundProfileImage>
        <HomepageProfileImage />
      </HomepageBackgroundProfileImage>
    </HomepageProfileImageContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  isSideBarHidden: selectIsSideBarHidden,
});

export default connect(mapStateToProps)(HomePageProfileImageContainer);
