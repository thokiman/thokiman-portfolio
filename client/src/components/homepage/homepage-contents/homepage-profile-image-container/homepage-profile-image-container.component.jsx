import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectIsSideBarActive } from '../../../../redux/header/header.selectors';
import HomepageProfileImageCartoon from '../homepage-profile-image-cartoon/homepage-profile-image-cartoon.component.jsx';
import {
  HomepageBackgroundProfileImage,
  HomepageProfileImage,
  HomepageProfileImageContainer,
} from './homepage-profile-image-container.styles';

export const HomePageProfileImageContainer = ({ isSideBarActive }) => {
  return (
    <HomepageProfileImageContainer $issidebaractive={isSideBarActive}>
      <HomepageProfileImageCartoon />
      <HomepageBackgroundProfileImage>
        <HomepageProfileImage />
      </HomepageBackgroundProfileImage>
    </HomepageProfileImageContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  isSideBarActive: selectIsSideBarActive,
});

export default connect(mapStateToProps)(HomePageProfileImageContainer);
