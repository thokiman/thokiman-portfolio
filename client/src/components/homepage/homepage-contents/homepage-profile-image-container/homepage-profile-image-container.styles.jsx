import styled, { css } from 'styled-components';
import HomepageProfile from '../../../../assets/profile/profile.png';

const blackColor = '#000000';

const maxWidthDesktop = 1366;
const maxWidthTablet = 1024;
const maxWidthSmartphone = 800;

const HomepageProfileImageIsNotActivated = css`
  display: none;
`;
const getSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return HomepageProfileImageIsNotActivated;
  return null;
};
export const HomepageProfileImageContainer = styled.div`
  height: 100%;
  width: 51.3vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  position: relative;
  background-color: ${blackColor};

  @media only screen and (max-width: ${maxWidthDesktop}px) {
    ${getSideBarActive}
    width: 100%;
    height: 250vh;
    background-color: ${blackColor};
  }
`;

export const HomepageBackgroundProfileImage = styled.div`
  width: 50vw;
  height: 100%;
  padding: 10px;
  display: flex;
  justify-content: center;
  align-items: center;

  @media only screen and (max-width: ${maxWidthDesktop}px) {
    display: block;

    width: 50vw;
    height: 120vh;
  }

  @media only screen and (max-width: ${maxWidthTablet}px) {
    height: 50vh;
  }

  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 100vw;
    height: 78vh;
  }
`;

export const HomepageProfileImage = styled.div`
  width: 100%;
  height: 100%;

  background-image: url(${HomepageProfile});
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;

  @media only screen and (max-width: ${maxWidthDesktop}px) {
    background-position: 100% 50%;
  }
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    background-position: 100% 85%;
  }
`;

HomepageProfileImageContainer.displayName = 'HomepageProfileImageContainer';
HomepageBackgroundProfileImage.displayName = 'HomepageBackgroundProfileImage';
HomepageProfileImage.displayName = 'HomepageProfileImage';
