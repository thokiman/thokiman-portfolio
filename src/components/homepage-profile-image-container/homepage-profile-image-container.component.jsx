import React from "react";

import {
  HomepageBackgroundProfileImage,
  HomepageProfileImage,
  HomepageProfileImageContainer,
} from "./homepage-profile-image-container.styles";

const HomePageProfileImageContainer = () => {
  return (
    <HomepageProfileImageContainer>
      <HomepageBackgroundProfileImage>
        <HomepageProfileImage />
      </HomepageBackgroundProfileImage>
    </HomepageProfileImageContainer>
  );
};

export default HomePageProfileImageContainer;
