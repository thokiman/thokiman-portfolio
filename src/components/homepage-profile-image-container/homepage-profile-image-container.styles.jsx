import styled from "styled-components";

import HomepageProfile from "../../assets/profile/profile.png";

export const HomepageProfileImageContainer = styled.div`
  height: 100%;
  max-width: 51.3vw;
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  background-color: #080808;
  @media only screen and (max-width: 800px) {
    display: none;
  }
`;

export const HomepageBackgroundProfileImage = styled.div`
  width: 900px;
  height: 100%;
  padding: 10px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const HomepageProfileImage = styled.div`
  width: 100%;
  height: 100%;
  background-image: url(${HomepageProfile});
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
`;

HomepageProfileImageContainer.displayName = "HomepageProfileImageContainer";
HomepageBackgroundProfileImage.displayName = "HomepageBackgroundProfileImage";
HomepageProfileImage.displayName = "HomepageProfileImage";
