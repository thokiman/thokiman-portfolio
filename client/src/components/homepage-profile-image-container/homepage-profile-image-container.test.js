import React from "react";
import { shallow } from "enzyme";
import { HomePageProfileImageContainer } from "./homepage-profile-image-container.component";

it("should render HomePageProfileImageContainer component", () => {
  expect(shallow(<HomePageProfileImageContainer />)).toMatchSnapshot();
});
