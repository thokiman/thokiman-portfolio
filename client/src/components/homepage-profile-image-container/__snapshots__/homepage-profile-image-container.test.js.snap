// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomePageProfileImageContainer component 1`] = `ShallowWrapper {}`;
